components/code_panel.js

```diff
@@ -1,35 +1,87 @@
 let {div, header, footer, button, p, hJSX} = require('@cycle/dom')
 import isolate from '@cycle/isolate';
 
+let tommy = require('tommy-the-runner')
+let Rx = require('rx')
+
 function intent({ DOM, context }) {
     let buttonClicks$ = DOM.select('.submit-button').events('click')
-        .map(ev => {
-            return new Date().getTime()
+        .flatMap(ev => {
+           var userCode = document.getElementById('user-code').value
+
+            // TODO: retrieve specs from the original JSON (store in context?)
+            var specs = `
+                            const expect = require('chai').expect
+                            const sum = require('subject')
+                            describe('sum', function () {
+                                it('should sum two numbers', function () {
+                                    expect(sum(1, 1)).to.equal(2)
+                                })
+                            })
+                        `
+
+            return Rx.Observable.fromPromise(tommy.run(userCode, specs)
+                .then(reporter => {
```

Comment:
since you are using `then` as `Rx.Observable::map`, you could leave promise-based api at `fromPromise` point and continue parsing with observables

`Rx.Observable.fromPromise(tommy.run(userCode, specs)).map(reporter => )`